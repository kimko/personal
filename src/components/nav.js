import React from "react";
import PropTypes from "prop-types";

export const Nav = ({ style, name, email, phone }) => {
  return (
    <section
      className="bb fixed dt w-100 cf"
      style={style}
      data-testid="test-section-nav"
    >
      <nav className="pl4 tracked">
        <div className="flex flex-wrap">
          <div className="w-100 w-third-ns">
            <h1 className="bl bw2 f5 pl1 mb0">{name}</h1>
          </div>
        </div>
        <div className="mv1 f6">
          <a href={`mailto:${email}`} className="db link dim" style={style}>
            {email}
          </a>

          <a href={`tel:${phone.raw}`} className="db link dim" style={style}>
            {phone.pretty}
          </a>
        </div>
      </nav>
    </section>
  );
};

Nav.propTypes = {
  style: PropTypes.object,
  name: PropTypes.string,
  email: PropTypes.string,
  phone: PropTypes.object,
};
