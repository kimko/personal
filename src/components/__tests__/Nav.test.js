import React from "react";
import { render, cleanup } from "@testing-library/react";

import { Nav } from "../nav";

afterEach(cleanup);

const name = "Angela Merkel";
const email = "kanzler@bundestag.de";
const phone = { raw: "+11234567890", pretty: "1-123-456-7890" };

it("renders a Nav section", () => {
  const { getByTestId, asFragment } = render(
    <Nav name={name} email={email} phone={phone} />,
  );

  expect(getByTestId("test-section-nav")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
