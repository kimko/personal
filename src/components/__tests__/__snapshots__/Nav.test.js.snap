// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Nav section 1`] = `
<DocumentFragment>
  <section
    class="bb fixed dt w-100 cf"
    data-testid="test-section-nav"
  >
    <nav
      class="pl4 tracked"
    >
      <div
        class="flex flex-wrap"
      >
        <div
          class="w-100 w-third-ns"
        >
          <h1
            class="bl bw2 f5 pl1 mb0"
          >
            Angela Merkel
          </h1>
        </div>
      </div>
      <div
        class="mv1 f6"
      >
        <a
          class="db link dim"
          href="mailto:kanzler@bundestag.de"
        >
          kanzler@bundestag.de
        </a>
        <a
          class="db link dim"
          href="tel:+11234567890"
        >
          1-123-456-7890
        </a>
      </div>
    </nav>
  </section>
</DocumentFragment>
`;
